fabfile.py

```diff
@@ -165,6 +180,198 @@ def listdir(path):
+    print json.dumps(data, indent=4)
+    return data
+
+
+@task
+def get_domain_list():
+    client = create_ovh_client()
+    result = client.get('/domain')
+    print json.dumps(result, indent=4)
+    return result
+
+
+@task
+def create_new_dns_record():
+    data = get_dns_entries_from_file()
+    if check_dns_entries(data) and check_domains(data):
+        client = create_ovh_client()
+        for domain in data:
+            for entry in data[domain]:
+                client.post('/domain/zone/{domain}/record'.format(domain=domain), fieldType=entry["type"], target=entry["value"])
+
+
+@task
+def create_new_dns_record_cmd_line_arguments(domain_entry, field_type_entry, target_entry):
+    client = create_ovh_client()
+    if domain_entry not in get_domain_list():
+        print("The domain {domain_entry} is not a valid domain".format(domain_entry=domain_entry))
+        print "The valid domains are:"
+        for domain in get_domain_list():
+            print domain
+    elif not check_field_type_exists(field_type_entry):
+        print("The field {field_type_entry} is not a valid field type".format(field_type_entry=field_type_entry))
+        print "Valid field types are A, AAAA, CNAME, DKIM, LOC, MX, NAPTR, NS, PTR, SPF, SRV, SSHFP, TXT"
+    else:
+        print("Creating new DNS record for the domain {domain_entry} with field type {field_type_entry} with target {target_entry}".format(
+            domain_entry=domain_entry, field_type_entry=field_type_entry, target_entry=target_entry))
+        client.post('/domain/zone/{domain_entry}/record'.format(domain_entry=domain_entry), fieldType=field_type_entry, target=target_entry)
+
+
+def create_ovh_client():
+    try:
+        import ovh
+    except:
+        print "Running pip install ovh"
+        local("pip install ovh")
+    client = ovh.Client(
+        endpoint=env.ovh_hosting__endpoint,
+        application_key=env.ovh_hosting__application_key,
+        application_secret=env.ovh_hosting__application_secret,
+        consumer_key=env.ovh_hosting__consumer_key,
+    )
+    return client
+
+
+@task
+def get_ovh_vps_snapshot():
+    client = create_ovh_client()
+    result = client.get('/vps/{vps_identifier}/snapshot'.format(vps_identifier=env.ovh_hosting__vps_identifier))
+    print json.dumps(result, indent=4)
+
+
+@task
+def set_ovh_vps_snapshot():
+    import datetime
+    today_date = datetime.datetime.now().strftime('%Y_%m_%d_%H_%M_%S')
+    client = create_ovh_client()
+    result = client.put('/vps/{vps_identifier}/snapshot'.format(vps_identifier=env.ovh_hosting__vps_identifier),
+                        description='snapshot-{today_date}'.format(today_date=today_date))
+    print json.dumps(result, indent=4)
+
+
+@task
+def list_secondary_dns_names_ovh():
+    client = create_ovh_client()
+    result = client.get('/vps/{vps_identifier}/secondaryDnsDomains'.format(vps_identifier=env.ovh_hosting__vps_identifier))
```

Comment:
For all of the `get` operations, if there is a failure, you should return the exception.